e2e/kube.go

```diff
@@ -76,32 +76,29 @@ func getClusterKubeconfigBytes(ctx context.Context, resourceGroupName, clusterNa
 }
 
 // this is a bit ugly, but we don't want to execute this piece concurrently with other tests
-func ensureDebugDaemonset(ctx context.Context, kube *Kubeclient) error {
-	manifest := getDebugDaemonset()
-	var ds v1.DaemonSet
-
-	if err := yaml.Unmarshal([]byte(manifest), &ds); err != nil {
-		return fmt.Errorf("failed to unmarshal debug daemonset manifest: %w", err)
+func ensureDebugDaemonsets(ctx context.Context, kube *Kubeclient) error {
+	manifests := getDebugDaemonsetManifests()
+	for _, manifest := range manifests {
+		if err := createDebugDeployment(ctx, kube, manifest); err != nil {
+			return err
```

Comment:
should we continue the loop in case only 1 DebugDeployment is failing ?

Reply:
We need both to perform all the checks, so i think its better to fail here